modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -374,6 +374,8 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
             self._info(f"Training will stop when the number of samples to pass reaches {self.num_samples_to_pass}.")
 
         if self._downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE:
+            # assertion since model validation by pydantic should catch this.
+            assert self._downsampler.supports_bts, "The downsampler does not support batch then sample"
```

Comment:
see my other comment on why I'd prefer to have this

Reply:
I don't get that yet - the if only checks the flag. For reasons outlined above (see new comment), we cannot trust the input here. And I would not silently not downsample in case we have invalid input. We should fail when we get invalid input